content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md

```diff
@@ -170,6 +170,33 @@ kubelet instance. A new kubelet instance deletes all the existing Unix sockets u
 `/var/lib/kubelet/device-plugins` when it starts. A device plugin can monitor the deletion
 of its Unix socket and re-register itself upon such an event.
 
+### Device plugin and unhealthy devices
+
+There are cases when devices fail or being shut down. Device Plugin responsibility
+in this case is to notify kubelet about this situation using the `ListAndWatchResponse` API.
+
+Once device is marked as unhealthy, kubelet will decrease the allocatable count
+for this resource on the Node to reflect how many devices can be used for scheduling new pods.
+Capacity count for the resource will not change.
+
+Pods that were assigned to the failed devices will continue be assigned to this device.
+It is typical that code relying on the device will start failing and Pod may get
+into Failed phase if `restartPolicy` for the Pod was not `Always` or enter the crash loop
+otherwise.
+
+Before the Kubernetes version 1.31, the way to know whether Pod is associated with the
+failed device was to use the [PodResources API](#monitoring-device-plugin-resources).
+
+{{< feature-state for_k8s_version="v1.31" state="alpha" >}}
```

Comment:
Consider using the "feature_gate_name" option for `feature-state` shortcode. This will make it more data-driven by pulling from the feature gate description file.

```suggestion
{{< feature-state feature_gate_name="ResourceHealthStatus" >}}
```